tests/console/commands/test_show.py

```diff
@@ -369,8 +370,331 @@ def test_show_outdated(app, poetry, installed, repo):
+                "hashes": {"cachy": [], "pendulum": [], "demo": [], "my-package": []},
+            },
+        }
+    )
+
+    tester.execute([("command", command.get_name()), ("--outdated", True)])
+
+    expected = """\
+cachy      0.1.0                       0.2.0                      
+my-package 0.1.1 ../project_with_setup 0.1.2 ../project_with_setup
+"""
+    actual = tester.get_display(True)
+
+    assert actual == expected
+
+
+@pytest.mark.parametrize("project_directory", ["project_with_git_dev_dependency"])
+def test_show_outdated_git_dev_dependency(app, poetry, installed, repo):
+    command = app.find("show")
+    tester = CommandTester(command)
+
+    cachy_010 = get_package("cachy", "0.1.0")
+    cachy_010.description = "Cachy package"
+    cachy_020 = get_package("cachy", "0.2.0")
+    cachy_020.description = "Cachy package"
+
+    pendulum_200 = get_package("pendulum", "2.0.0")
+    pendulum_200.description = "Pendulum package"
+
+    installed.add_package(cachy_010)
+    installed.add_package(pendulum_200)
+
+    repo.add_package(cachy_010)
+    repo.add_package(cachy_020)
+    repo.add_package(pendulum_200)
+
+    poetry.locker.mock_lock_data(
+        {
+            "package": [
+                {
+                    "name": "cachy",
+                    "version": "0.1.0",
+                    "description": "Cachy package",
+                    "category": "main",
+                    "optional": False,
+                    "platform": "*",
+                    "python-versions": "*",
+                    "checksum": [],
+                },
+                {
+                    "name": "pendulum",
+                    "version": "2.0.0",
+                    "description": "Pendulum package",
+                    "category": "main",
+                    "optional": False,
+                    "platform": "*",
+                    "python-versions": "*",
+                    "checksum": [],
+                },
+                {
+                    "name": "demo",
+                    "version": "0.1.2",
+                    "description": "Demo package",
+                    "category": "dev",
+                    "optional": False,
+                    "platform": "*",
+                    "python-versions": "*",
+                    "checksum": [],
+                    "source": {
+                        "type": "git",
+                        "reference": "9cf87a285a2d3fbb0b9fa621997b3acc3631ed24",
+                        "url": "https://github.com/demo/pyproject-demo.git",
+                    },
+                },
+            ],
+            "metadata": {
+                "python-versions": "*",
+                "platform": "*",
+                "content-hash": "123456789",
+                "hashes": {"cachy": [], "pendulum": [], "demo": []},
+            },
+        }
+    )
+
+    tester.execute([("command", command.get_name()), ("--outdated", True)])
+
+    expected = """\
+cachy 0.1.0 0.2.0 Cachy package
```

Comment:
Shouldn't we show the revisions of both versions? That way it makes it clear that they are Git dependencies.

Reply:
Do you mean set the locked version of `demo` to `0.1.1` to test the output?

Reply:
Yes, so that we can test that the result of `show` is similar to the output of `install` for git dependencies, i.e.:

```
demo 0.1.2 9cf87a28 0.2.0 other-hash Demo package
```

Reply:
Ok. I'll also update the other git test to have an outdated dependency to ensure it won't show when `--no-dev` is passed.

Reply:
I added a commit to address your feedback. Thanks!